base/loading.jl

```diff
@@ -591,9 +591,15 @@ function explicit_manifest_uuid_path(project_file::String, pkg::PkgId)::Union{No
             return path
         end
         hash === nothing && return nothing
+        # Look for bundled packages in `$project/packages`
+        depot_path = DEPOT_PATH
+        project = active_project()
+        if project !== nothing
+            depot_path = [abspath(dirname(project)); depot_path]
```

Comment:
Can you move it to a sub-directory? Quoting my other comments:

> Does it make sense to use `.julia` as the parent directory (i.e., `/path_to_active_project/.julia/artifacts` and `/path_to_active_project/.julia/packages`), to avoid cluttering the package directory? I think there are other things you might want to make project-local. For example, it would also be nice if the system image is automatically loaded from, e.g., `/path_to_active_project/.julia/sys.so` [JuliaLang/julia#33973 (comment)](https://github.com/JuliaLang/julia/issues/33973#issuecomment-601541817)
>
> ---  https://github.com/JuliaLang/Pkg.jl/pull/1727#issuecomment-601994269

What I'm suggesting is something like:

```suggestion
            depot_path = [abspath(dirname(project), ".julia"); depot_path]
```

`".julia"` is an example. Alternatives: https://github.com/JuliaLang/Pkg.jl/pull/1727#issuecomment-601997518